src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/WorkingDaysService.kt

```diff
@@ -7,29 +7,72 @@ import java.time.LocalDate
 @Service
 class WorkingDaysService(private val bankHolidayService: BankHolidayService) {
 
-  fun addWorkingDays(date: LocalDate, workingDays: Int): LocalDate {
-    val bankHolidays = bankHolidayService.getBankHolidaysForEnglandAndWales()
+  fun addWorkingDays(date: LocalDate, workingDays: Int, isEarlyReleaseWeekend: Boolean = false): LocalDate {
     var adjustedDate = date
-    for (i in 0 until workingDays) {
-      adjustedDate = getNextWorkingDay(bankHolidays, adjustedDate)
+    repeat(workingDays) {
+      adjustedDate = getNextWorkingDay(adjustedDate, isEarlyReleaseWeekend)
     }
     return adjustedDate
   }
 
-  fun isWeekend(date: LocalDate): Boolean {
-    val weekend = listOf(DayOfWeek.SATURDAY, DayOfWeek.SUNDAY)
+  fun subWorkingDays(date: LocalDate, workingDays: Int, isEarlyReleaseWeekend: Boolean = false): LocalDate {
+    var adjustedDate = date
+    repeat(workingDays) {
+      adjustedDate = getPreviousWorkingDay(adjustedDate, isEarlyReleaseWeekend)
+    }
+    return adjustedDate
+  }
+
+  fun getWorkingDaysRange(
```

Comment:
Is there a requirement for this method?

Reply:
This is so we can generate the sequence of dates needed for function such as `getCutOffDateForLicenceTimeOut` in the ReleaseDateService. So instead of:

` fun getCutOffDateForLicenceTimeOut(now: Clock? = null): LocalDate {
    return generateSequence(LocalDate.now(now ?: clock)) { it.plusDays(1) }
      .filterNot { isBankHolidayOrWeekend(it) }
      .take(maxNumberOfWorkingDaysToUpdateLicenceTimeOutStatus)
      .last()
  }`
  
  we would have
  
      `return workingDaysService.getWorkingDaysRange(
      LocalDate.now(now ?: clock),
      maxNumberOfWorkingDaysToUpdateLicenceTimeOutStatus,
    ).last()`